### PR TITLE
[tests] Make BrokerClientIntegrationTest testing behavior deterministic

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

The test is flaky.

```
2018-09-14\T\17:46:29.848 [ERROR] testUnsupportedBatchMessageConsumer(org.apache.pulsar.client.impl.BrokerClientIntegrationTest)  Time elapsed: 3.161 s  <<< FAILURE!
java.lang.AssertionError: Received message my-message-5 did not match the expected message my-message-0 expected [my-message-0] but found [my-message-5]
        at org.testng.Assert.fail(Assert.java:96)
        at org.testng.Assert.failNotEquals(Assert.java:776)
        at org.testng.Assert.assertEqualsImpl(Assert.java:137)
        at org.testng.Assert.assertEquals(Assert.java:118)
        at org.apache.pulsar.client.api.ProducerConsumerBase.testMessageOrderAndDuplicates(ProducerConsumerBase.java:51)
        at org.apache.pulsar.client.impl.BrokerClientIntegrationTest.testUnsupportedBatchMessageConsumer(BrokerClientIntegrationTest.java:357)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
        at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

The problem is due to we used a `time` based batching policy in testing. There is no guarantee how messages can be batched, hence
the ordering and duplication check can fail on shared subscription

*Changes*

Set batching delay to a very large value and make sure the messages are in one batch.

### CODE REVIEW

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
retest this please

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java [hunk 1]
```diff
@@ -333,12 +336,13 @@ public void testUnsupportedBatchMessageConsumer(SubscriptionType subType) throws
         // verification
         consumer1.setClientCnx(null);
         // (2) send batch-message which should not be able to consume: as broker will disconnect the consumer
-        for (int i = 0; i < 10; i++) {
+        CompletableFuture<MessageId> lastSendFuture = null;
+        for (int i = 0; i < numMessagesPerBatch; i++) {
             String message = "my-message-" + i;
-            batchProducer.sendAsync(message.getBytes());
+            lastSendFuture = batchProducer.sendAsync(message.getBytes());
         }
```
Comment (ivankelly) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java]: you could use producer.flush() here.

**Comment by sijie** [comment]
retest this please

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java (hunk 1 above)
Comment (sijie) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java]: changed to use producer.flush

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by ivankelly** [comment]
rerun integration tests
rerun java8 tests

**Pull request merged**